crates/bevy_gizmos/src/aabb.rs

```diff
@@ -108,7 +109,7 @@ fn color_from_entity(entity: Entity) -> LegacyColor {
     const RATIO_360: f32 = 360.0 / u32::MAX as f32;
     let hue = index.wrapping_mul(FRAC_U32MAX_GOLDEN_RATIO) as f32 * RATIO_360;
 
-    LegacyColor::hsl(hue, 1., 0.5)
+    Hsla::hsl(hue, 1., 0.5).into()
```

Comment:
I think we should consider adding friendly constructors for Color like we have currently. I think this:

```rust
Color::hsl(hue, 1.0, 0.5)
```

Reads (and writes) much better than

```rust
Hsla::hsl(hue, 1.0, 0.5).into()
```

Reply:
This would also serve to reduce breakage provided our methods are 1:1 (and i think they probably can / should be)

Reply:
(I'm not saying we should make "not breaking people" a priority. We should only add what makes sense)

Reply:
I agree with this. Since we've delegated domain-specific knowledge of each space to its own type all 20 of these functions (10 spaces with constructors with/out alpha) would be of the form:

```rust
impl Color {
    pub const fn rgb(r: f32, g: f32, b: f32) -> Self {
        Self::Srgba(Srgba::rgb(r, g, b))
    }
}
```

Reply:
I agree. Since all the domain specific information about the various colour spaces are confined to their respective types, it should keep the `Color` type reasonably clean. It'll basically be 20 or so `const` functions:

```rust
impl Color {
    pub const fn rgb(r: f32, g: f32, b: f32) -> Self {
        Self::Srgba(Srgba::rgb(r, g, b)) 
    }
}
```

Reply:
Yep, can do!

Reply:
Split into https://github.com/bevyengine/bevy/pull/12171 :)